timemachine/md/minimizer.py

```diff
@@ -535,40 +618,28 @@ def local_minimize(
     def val_and_grad_fn_local(x_local):
         x_prime = x0.copy()
         x_prime[free_idxs] = x_local
-        u_full, grad_full = minimizer_val_and_grad(x_prime)
+        U_full, grad_full = minimizer_val_and_grad(x_prime)
         # The GPU Potentials can return NaN if value would have overflowed in uint64
-        if np.isnan(u_full):
-            u_full = np.inf
+        # FIRE only looks at the gradients and can recover from NaN energies
+        if method != "FIRE" and np.isnan(U_full):
+            U_full = np.inf
             grad_full = np.nan * grad_full
```

Comment:
Docstring wording: I don't think it can "recover from" NaN energies, it just doesn't look at energies (which is a benefit in cases where energy is nan, but forces are still accurate)

Reply:
Clarified in https://github.com/proteneer/timemachine/pull/1405/commits/b2d9e5136c786f19191e8cb79504fdfcea6e9287